plugin/pkg/scheduler/core/scheduling_queue_test.go

```diff
@@ -87,12 +88,16 @@ func TestPriorityQueue_Add(t *testing.T) {
 
 func TestPriorityQueue_Pop(t *testing.T) {
 	q := NewPriorityQueue()
+	wg := sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		if p, err := q.Pop(); err != nil || p != &highPriorityPod {
 			t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPod.Name, p.Name)
 		}
 	}()
 	q.Add(&highPriorityPod)
+	wg.Wait()
```

Comment:
If you want to change the test to ensure that `Pop()` is always executed before `Add()`, you should place `wg.Wait()` before `q.Add(...)`.

Reply:
If `wg.Wait()` is put before `q.Add(...)`, it will cause dead lock.
This change makes sure that `q.Pop()` is `runned` and `finished` successfully.

Reply:
Ah, right. Since Pop() is tested in other tests, my intention for writing this test was to ensure that when Pop() is executed before Add(), it waits and returns what Add() adds to the queue. Your change won't ensure that, but it is still useful and ensures that Pop() is executed.